routing/delegated.go

```diff
@@ -147,13 +151,62 @@ func parse(visited map[string]bool,
 	return router, nil
 }
 
-type ExtraReframeParams struct {
+type ExtraHTTPParams struct {
 	PeerID     string
 	Addrs      []string
 	PrivKeyB64 string
 }
 
-func reframeRoutingFromConfig(conf config.Router, extraReframe *ExtraReframeParams) (routing.Routing, error) {
+func httpRoutingFromConfig(conf config.Router, extraHTTP *ExtraHTTPParams) (routing.Routing, error) {
+	params := conf.Parameters.(*config.HTTPRouterParams)
+	if params.Endpoint == "" {
+		return nil, NewParamNeededErr("Endpoint", conf.Type)
+	}
+
+	params.FillDefaults()
+
+	// Increase per-host connection pool since we are making lots of concurrent requests.
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.MaxIdleConns = 500
+	transport.MaxIdleConnsPerHost = 100
+
+	delegateHTTPClient := &http.Client{
```

Comment:
will need to wrap this in https://github.com/ipfs/go-libipfs/blob/main/routing/http/client/transport.go#L9  but I can take care of that in a followup PR